runtime/src/iree/io/formats/gguf/gguf_parser_test.cc

```diff
@@ -13,43 +13,49 @@
 namespace iree {
 namespace {
 
-static iree_io_file_handle_t* OpenTestFile(const char* name) {
-  const struct iree_file_toc_t* file_toc = iree_io_gguf_files_create();
-  for (size_t i = 0; i < iree_io_gguf_files_size(); ++i) {
-    if (strcmp(file_toc[i].name, name) == 0) {
-      iree_io_file_handle_t* file_handle = NULL;
-      IREE_CHECK_OK(iree_io_file_handle_wrap_host_allocation(
-          IREE_IO_FILE_ACCESS_READ,
-          iree_make_byte_span((void*)file_toc[i].data, file_toc[i].size),
-          iree_io_file_handle_release_callback_null(), iree_allocator_system(),
-          &file_handle));
-      return file_handle;
+class GgufFormatTest : public ::testing::TestWithParam<std::string> {
+ protected:
+  iree_io_file_handle_t* OpenTestFile(const char* name) {
+    std::string parameterized_name = std::string(name) + GetParam() + ".gguf";
+
+    const struct iree_file_toc_t* file_toc = iree_io_gguf_files_create();
+    for (size_t i = 0; i < iree_io_gguf_files_size(); ++i) {
+      if (strcmp(file_toc[i].name, parameterized_name.c_str()) == 0) {
+        iree_io_file_handle_t* file_handle = NULL;
+        IREE_CHECK_OK(iree_io_file_handle_wrap_host_allocation(
+            IREE_IO_FILE_ACCESS_READ,
+            iree_make_byte_span((void*)file_toc[i].data, file_toc[i].size),
+            iree_io_file_handle_release_callback_null(),
+            iree_allocator_system(), &file_handle));
+        return file_handle;
+      }
     }
+    IREE_CHECK_OK(
+        iree_make_status(IREE_STATUS_NOT_FOUND,
+                         "test file `%s` not found embedded into test binary",
+                         parameterized_name));
+    return NULL;
   }
-  IREE_CHECK_OK(iree_make_status(
-      IREE_STATUS_NOT_FOUND,
-      "test file `%s` not found embedded into test binary", name));
-  return NULL;
-}
+};
 
-TEST(GgufFormatTest, Empty) {
+TEST_P(GgufFormatTest, Empty) {
   iree_io_parameter_index_t* index = NULL;
   IREE_ASSERT_OK(
       iree_io_parameter_index_create(iree_allocator_system(), &index));
 
-  iree_io_file_handle_t* file_handle = OpenTestFile("empty.gguf");
+  iree_io_file_handle_t* file_handle = OpenTestFile("empty");
```

Comment:
not the biggest fan of obscuring things here as it makes it harder to easily know what file is being read by inspection

Reply:
Okay, thanks. Will keep the tests simple.